clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -34,30 +34,28 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
-      return LifeCycleOpCodec.encode((LifecycleMessage) message);
-    } else {
-      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
-    }
+    return message.encode();
```

Comment:
Messages only know about their structure of data, so to me I think it makes sense to have encode/decode in messages itself. Also the encode/decode of messages is only done by platform using the codec interface, so if the message is already there, just call encode on that message.